test/inductor/test_auto_chunker.py

```diff
@@ -176,6 +183,74 @@ def f(x, y):
                 f"Actual peak_memory {peak_memory}, expected bound {expected_bound}",
             )
 
+    @config.patch("auto_chunker.num_chunk", config.auto_chunker.num_chunk or 16)
+    @largeTensorTest("12GB", device=GPU_TYPE, inductor=True)
+    @parametrize("gradient_accumulation_steps", [1, 2])
+    def test_gradient_accumulation(self, gradient_accumulation_steps):
+        B = 32
+        T = 1024
+        C = 768
+        V = 50257
+
+        dtype = torch.bfloat16
+
+        mod = LinearAndCEL(C, V).cuda().to(dtype)
+
+        def f(x, y):
+            x.grad = None
+
+            x = x * 2
+            loss = mod(x, y) / gradient_accumulation_steps
+            loss.backward()
+            return loss
+
+        def step(func, xs, ys):
+            mod.linear.weight.grad = None
+            mod.linear.bias.grad = None
+
+            tot = 0
+            for x, y in zip(xs, ys):
+                loss = func(x, y)
+                tot += loss.detach().item()
+            return tot
+
+        opt_f = torch.compile(f)
+
+        xs = [
+            torch.randn(B, T, C, dtype=dtype, requires_grad=True, device="cuda")
+            for _ in range(gradient_accumulation_steps)
+        ]
+        ys = [
+            torch.randint(0, V, (B, T)).cuda()
+            for _ in range(gradient_accumulation_steps)
+        ]
+
+        expect = (
+            step(f, xs, ys),
+            *[x.grad for x in xs],
+            mod.linear.weight.grad,
+            mod.linear.bias.grad,
+        )
+        torch.cuda.reset_peak_memory_stats()
+        actual = (
+            step(opt_f, xs, ys),
+            *[x.grad for x in xs],
+            mod.linear.weight.grad,
+            mod.linear.bias.grad,
+        )
+        peak_memory = torch.cuda.max_memory_allocated()
+        print(f"Peak memory {peak_memory / 10**9:.6f} GB")
+
+        self.assertTrue(same(expect, actual, tol=1e-3), f"{expect=}\n{actual=}")
+
+        if config.auto_chunker.enable:
```

Comment:
shouldnt this always be enabled in this test?

Reply:
I occasionally run this test with AutoChunker disabled leveraging this decorator in this test class:
```
@config.patch(
    "auto_chunker.enable", os.environ.get("TORCHINDUCTOR_AUTO_CHUNKER", "1") == "1"
)
```

That's mainly used in early development to check compile behavior without chunking though.